### PR TITLE
Add CLI version switching

### LINKED ISSUE #538: Add Cadence 1.0 Mode
### Issue To Be Solved

With changes to how Cadence 1.0 installations are managed coming to the Flow CLI (& a standardized command name) https://github.com/onflow/flow-cli/pull/1410, it is only intuitive that we add support in VSCode to easily switch to this setting.

Currently, using Cadence 1.0 in VSCode means either replacing your pre-C1.0 CLI or specifying a custom `cadence.flowCommand`

This can now be achieved by adding a Cadence 1.0 mode which defaults to `flow-c1` instead of `flow` command inside of VSCode.  

### (Optional): Suggest A Solution

 - Add a setting in VSCode to switch to C1.0.
 - Update messaging about how to use C1.0 in CLI.
 - Potentially add UI piece that helps with this (i.e. at bottom of VSCode window)

### PR DESCRIPTION
**Pull request by jribbink**
Closes #538 

## Description

Adds ability to select versions of the CLI with native support for `flow-c1`

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by jribbink** [comment]
tests pass locally - there are seemingly issues with `@vscode/test-electron` that break these, will be addressed subsequently (probably need to update)

File: extension/src/server/flow-config.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ export class FlowConfig implements Disposable {
 
   // Watch and reload flow configuration when changed.
   #watchWorkspaceConfiguration (): Subscription {
-    return this.#settings.watch$(config => config.customConfigPath).subscribe(() => {
+    return this.#settings.settings$(config => config.customConfigPath).subscribe(() => {
```
Comment (bthaile) [inline extension/src/server/flow-config.ts:175]: settings.settings ? this look weird, just pointing it out if it's a typo

**Review by bthaile — APPROVED** [review]
lgtm

File: extension/src/server/language-server.ts [hunk 2]
```diff
@@ -3,34 +3,37 @@ import { window } from 'vscode'
...
+    this.#config = config
   }
 
   async activate (): Promise<void> {
     if (this.isActive) return
     await this.deactivate()
 
     this.#isActive = true
-    await this.startClient()
+    await this.startClient().catch(() => {})
```
Comment (bthaile) [inline extension/src/server/language-server.ts:36]: would an error message here get swelled?

**Comment by bthaile** [comment]
How does the correct language server get chosen based on the version of flow-cli used?

File: extension/src/server/flow-config.ts (hunk 1 above)
Comment (jribbink) [reply extension/src/server/flow-config.ts:175]: Yeah, I'll switch the back I think.  I was trying to follow the pattern with the other observables where we have `XX.value$` for the observable and `XX.getValue()` to get the current value, but it's probably a bit confusing for settings.

File: extension/src/server/language-server.ts (hunk 2 above)
Comment (jribbink) [reply extension/src/server/language-server.ts]: Yeah good callout.  I'll fix this & add comments as to why exactly this error is caught.

**Comment by jribbink** [comment]
> How does the correct language server get chosen based on the version of flow-cli used?

The `LanguageServerAPI` watches for the binary to change from the `CliProvider` and will restart accordingly.  I realized it maybe wasn't super clear, so I've moved some stuff around and added some more comments to make it more obvious that this is how it works.

**Pull request merged**